src/service/resources-impl.js

```diff
@@ -224,6 +232,50 @@ export class Resources {
   }
 
   /**
+   * An ugly fix for IE's problem with `matchMedia` API, where media queries
+   * are evaluated incorrectly. See #2577 for more details.
+   * @param {!Window} win
+   * @private
+   */
+  fixMediaIe_(win) {
+    if (!this.platform_.isIe() || this.matchMediaIeQuite_(win)) {
+      this.relayoutAll_ = true;
+      return;
+    }
+
+    // Poll until the expression resolves correctly, but only up to a point.
+    const endTime = timer.now() + 2000;
+    const interval = win.setInterval(() => {
```

Comment:
There are a few other polling places throughout the code base. Might be time for a poll helper, with optional `maxWait`?